pytorch_lightning/plugins/training_type/dp.py

```diff
@@ -22,6 +23,18 @@
 from pytorch_lightning.utilities.apply_func import apply_to_collection
 
 
+def _ignore_scalar_return_in_dp():
```

Comment:
wouldn't it be better to have it where the overrides are, i.e., in pytorch_lightning.overrides?

Reply:
I personally prefer it where it is.

Overrides seems more to me for full class/function overrides

Reply:
To be clear, the warning is produced by nn.DataParallel which we wrap/override in the files in pl/overrides. The plugin here has nothing to do with that and in fact we have another plugin that uses DataParallel, the DDP2Plugin. So you would have to copy the ignore there too :)

Reply:
touché then

Reply:
@awaelchli @carmocca I'll move it to the overrides